dll/win32/browseui/CAutoComplete.cpp

```diff
@@ -272,16 +272,27 @@ EditWordBreakProcW(LPWSTR lpch, INT index, INT count, INT code)
         {
             if (index)
                 --index;
-            while (index && !IsWordBreak(lpch[index]))
+            while (index)
+            {
+                if (!IsWordBreak(lpch[index]) && IsWordBreak(lpch[index-1]))
+                    break;
                 --index;
+            }    
             return index;
         }
         case WB_RIGHT:
         {
             if (!count)
                 break;
-            while (index < count && lpch[index] && !IsWordBreak(lpch[index]))
+            while (index < count && lpch[index])
+            {
+                if (IsWordBreak(lpch[index]) && !IsWordBreak(lpch[index+1]))
```

Comment:
Suppose the lpch buffer is the following string: `abcde`, so we have: `count == 5` (if that parameter does not count the possible NUL-terminator of the string; this is not clear at all from the [MSDN article](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nc-winuser-editwordbreakprocw) ...)
The `index` thus can go from 0 to 4 included.
If `index` happens to be == 4 (it's still `< count`), `lpch[index+1]` is potentially reading outside of the string, and the second `IsWordBreak` check would behave weirdly, to say the least.